python312Packages.craft-providers: 1.24.0 -> 1.24.1

r-ryantm opened the pull request:
Automatic update generated by [nixpkgs-update](https://github.com/ryantm/nixpkgs-update) tools. This update was made based on information from passthru.updateScript.


meta.description for python312Packages.craft-providers is: Interfaces for instantiating and controlling a variety of build environments

meta.homepage for python312Packages.craft-providers is: https://github.com/canonical/craft-providers

meta.changelog for python312Packages.craft-providers is: https://github.com/canonical/craft-providers/releases/tag/1.24.1


###### Updates performed
- Ran passthru.UpdateScript

###### To inspect upstream changes





###### Impact

<b>Checks done</b>

---

- built on NixOS
- The tests defined in `passthru.tests`, if any, passed
- found 1.24.1 with grep in /nix/store/fvym54gpgkd00hil575lj6b9jjhqz45x-python3.12-craft-providers-1.24.1
- found 1.24.1 in filename of file in /nix/store/fvym54gpgkd00hil575lj6b9jjhqz45x-python3.12-craft-providers-1.24.1

---

<details>
<summary>
<b>Rebuild report</b> (if merged into master) (click to expand)
</summary>

```
10 total rebuild path(s)

10 package rebuild(s)

First fifty rebuilds by attrpath

charmcraft
python311Packages.craft-application
python311Packages.craft-application-1
python311Packages.craft-providers
python312Packages.craft-application
python312Packages.craft-application-1
python312Packages.craft-providers
rockcraft
snapcraft
```

</details>

<details>
<summary>
<b>Instructions to test this update</b> (click to expand)
</summary>

---

Either **download from Cachix**:
```
nix-store -r /nix/store/fvym54gpgkd00hil575lj6b9jjhqz45x-python3.12-craft-providers-1.24.1 \
  --option binary-caches 'https://cache.nixos.org/ https://nix-community.cachix.org/' \
  --option trusted-public-keys '
  nix-community.cachix.org-1:mB9FSh9qf2dCimDSUo8Zy7bkq5CX+/rkCWyvRCYg3Fs=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(The Cachix cache is only trusted for this store-path realization.)
For the Cachix download to work, your user must be in the `trusted-users` list or you can use `sudo` since root is effectively trusted.

Or, **build yourself**:
```
nix-build -A python312Packages.craft-providers https://github.com/r-ryantm/nixpkgs/archive/5f19e185d06974ad7518437eea17f4011ae3050a.tar.gz
```
Or:
```
nix build github:r-ryantm/nixpkgs/5f19e185d06974ad7518437eea17f4011ae3050a#python312Packages.craft-providers
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/fvym54gpgkd00hil575lj6b9jjhqz45x-python3.12-craft-providers-1.24.1
ls -la /nix/store/fvym54gpgkd00hil575lj6b9jjhqz45x-python3.12-craft-providers-1.24.1/bin
```

---

</details>
<br/>



### Pre-merge build results

We have automatically built all packages that will get rebuilt due to
this change.

This gives evidence on whether the upgrade will break dependent packages.
Note sometimes packages show up as _failed to build_ independent of the
change, simply because they are already broken on the target branch.

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages failed to build:</summary>
  <ul>
    <li>charmcraft</li>
    <li>charmcraft.dist</li>
    <li>rockcraft</li>
    <li>rockcraft.dist</li>
    <li>snapcraft</li>
    <li>snapcraft.dist</li>
  </ul>
</details>
<details>
  <summary>12 packages built:</summary>
  <ul>
    <li>python311Packages.craft-application</li>
    <li>python311Packages.craft-application-1</li>
    <li>python311Packages.craft-application-1.dist</li>
    <li>python311Packages.craft-application.dist</li>
    <li>python311Packages.craft-providers</li>
    <li>python311Packages.craft-providers.dist</li>
    <li>python312Packages.craft-application</li>
    <li>python312Packages.craft-application-1</li>
    <li>python312Packages.craft-application-1.dist</li>
    <li>python312Packages.craft-application.dist</li>
    <li>python312Packages.craft-providers</li>
    <li>python312Packages.craft-providers.dist</li>
  </ul>
</details>

---

###### Maintainer pings

cc @jnsgruk for [testing](https://github.com/ryantm/nixpkgs-update/blob/main/doc/nixpkgs-maintainer-faq.md#r-ryantm-opened-a-pr-for-my-package-what-do-i-do).

r-ryantm committed:
python312Packages.craft-providers: 1.24.0 -> 1.24.1

jnsgruk commented:
Result of `nixpkgs-review pr 325601` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages failed to build:</summary>
  <ul>
    <li>charmcraft</li>
    <li>charmcraft.dist</li>
    <li>rockcraft</li>
    <li>rockcraft.dist</li>
    <li>snapcraft</li>
    <li>snapcraft.dist</li>
  </ul>
</details>
<details>
  <summary>12 packages built:</summary>
  <ul>
    <li>python311Packages.craft-application</li>
    <li>python311Packages.craft-application-1</li>
    <li>python311Packages.craft-application-1.dist</li>
    <li>python311Packages.craft-application.dist</li>
    <li>python311Packages.craft-providers</li>
    <li>python311Packages.craft-providers.dist</li>
    <li>python312Packages.craft-application</li>
    <li>python312Packages.craft-application-1</li>
    <li>python312Packages.craft-application-1.dist</li>
    <li>python312Packages.craft-application.dist</li>
    <li>python312Packages.craft-providers</li>
    <li>python312Packages.craft-providers.dist</li>
  </ul>
</details>

jnsgruk commented:
This should be held until I can iron out the issues in snapcraft/rockcraft etc.

I'll try to get to it over the next few days.

jnsgruk committed:
python3Packages.keyring_24: init at 24.3.1

Version 25 of `keyring` introduced a breaking change by removing
a compatibility API. Several packages, including craft-store, rely
upon this API.

This is a copy of the existing `keyring` package, pinned to the last
version in the 24 series.

jnsgruk committed:
python3Packages.craft-store: use `keyring_24` instead of `keyring`.

jnsgruk committed:
python3Packages.craft-store: relax `macaroonbakery` dep version

jnsgruk commented:
This can be closed once #325807 is merged.

jnsgruk commented:
Result of `nixpkgs-review pr 325601` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages failed to build:</summary>
  <ul>
    <li>charmcraft</li>
    <li>charmcraft.dist</li>
    <li>rockcraft</li>
    <li>rockcraft.dist</li>
    <li>snapcraft</li>
    <li>snapcraft.dist</li>
  </ul>
</details>
<details>
  <summary>20 packages built:</summary>
  <ul>
    <li>python311Packages.craft-application</li>
    <li>python311Packages.craft-application-1</li>
    <li>python311Packages.craft-application-1.dist</li>
    <li>python311Packages.craft-application.dist</li>
    <li>python311Packages.craft-providers</li>
    <li>python311Packages.craft-providers.dist</li>
    <li>python311Packages.craft-store</li>
    <li>python311Packages.craft-store.dist</li>
    <li>python311Packages.keyring-24</li>
    <li>python311Packages.keyring-24.dist</li>
    <li>python312Packages.craft-application</li>
    <li>python312Packages.craft-application-1</li>
    <li>python312Packages.craft-application-1.dist</li>
    <li>python312Packages.craft-application.dist</li>
    <li>python312Packages.craft-providers</li>
    <li>python312Packages.craft-providers.dist</li>
    <li>python312Packages.craft-store</li>
    <li>python312Packages.craft-store.dist</li>
    <li>python312Packages.keyring-24</li>
    <li>python312Packages.keyring-24.dist</li>
  </ul>
</details>

jnsgruk committed:
rockcraft: use `setuptools.dist` rather than `distutils.util`

jnsgruk committed:
charmcraft: use `setuptools.dist` rather than `distutils.util`

jnsgruk committed:
python3Packages.launchpadlib: use `keyring_24` dependency to avoid clashes

jnsgruk committed:
snapcraft: 8.2.0 -> 8.2.12

jnsgruk committed:
snapcraft: run `nixfmt-rfc-style`

jnsgruk committed:
charmcraft: run `nixfmt-rfc-style`

jnsgruk commented:
@drupol this one got a bit big, so while I'd normally merge these, it seems like the right thing for someone else to take a look!

Mostly version and style bumps - only other change is the introduction of `keyring-24`.

jnsgruk commented:
Result of `nixpkgs-review pr 325601` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>devpi-client</li>
    <li>devpi-client.dist</li>
  </ul>
</details>
<details>
  <summary>58 packages built:</summary>
  <ul>
    <li>breezy (python312Packages.breezy)</li>
    <li>breezy.dist (python312Packages.breezy.dist)</li>
    <li>cabal2nix</li>
    <li>charmcraft</li>
    <li>charmcraft.dist</li>
    <li>dep2nix</li>
    <li>flatpak-builder</li>
    <li>flatpak-builder.doc</li>
    <li>flatpak-builder.installedTests</li>
    <li>flatpak-builder.man</li>
    <li>hydra_unstable</li>
    <li>luarocks-packages-updater</li>
    <li>nix-prefetch-bzr</li>
    <li>nix-prefetch-scripts</li>
    <li>nix-update-source</li>
    <li>nix-update-source.dist</li>
    <li>python311Packages.breezy</li>
    <li>python311Packages.breezy.dist</li>
    <li>python311Packages.check-manifest</li>
    <li>python311Packages.check-manifest.dist</li>
    <li>python311Packages.craft-application</li>
    <li>python311Packages.craft-application-1</li>
    <li>python311Packages.craft-application-1.dist</li>
    <li>python311Packages.craft-application.dist</li>
    <li>python311Packages.craft-archives</li>
    <li>python311Packages.craft-archives.dist</li>
    <li>python311Packages.craft-providers</li>
    <li>python311Packages.craft-providers.dist</li>
    <li>python311Packages.craft-store</li>
    <li>python311Packages.craft-store.dist</li>
    <li>python311Packages.keyring-24</li>
    <li>python311Packages.keyring-24.dist</li>
    <li>python311Packages.launchpadlib</li>
    <li>python311Packages.launchpadlib.dist</li>
    <li>python312Packages.check-manifest</li>
    <li>python312Packages.check-manifest.dist</li>
    <li>python312Packages.craft-application</li>
    <li>python312Packages.craft-application-1</li>
    <li>python312Packages.craft-application-1.dist</li>
    <li>python312Packages.craft-application.dist</li>
    <li>python312Packages.craft-archives</li>
    <li>python312Packages.craft-archives.dist</li>
    <li>python312Packages.craft-providers</li>
    <li>python312Packages.craft-providers.dist</li>
    <li>python312Packages.craft-store</li>
    <li>python312Packages.craft-store.dist</li>
    <li>python312Packages.keyring-24</li>
    <li>python312Packages.keyring-24.dist</li>
    <li>python312Packages.launchpadlib</li>
    <li>python312Packages.launchpadlib.dist</li>
    <li>rockcraft</li>
    <li>rockcraft.dist</li>
    <li>silver-platter</li>
    <li>silver-platter.dist</li>
    <li>snapcraft</li>
    <li>snapcraft.dist</li>
    <li>vcstool</li>
    <li>vcstool.dist</li>
  </ul>
</details>

jnsgruk commented:
The failure for `devpi-client` seems unrelated - it seems to be missing `packaging-legacy` in the runtime dependencies, which doesn't appear to be packaged in nix.

SuperSandro2000 commented on pkgs/top-level/python-packages.nix:
```diff
@@ -6483,6 +6483,8 @@ self: super: with self; {
 
   keyring = callPackage ../development/python-modules/keyring { };
 
+  keyring-24 = callPackage ../development/python-modules/keyring-24 { };
```
```suggestion
  keyring_24 = callPackage ../development/python-modules/keyring_24 { };
```

jnsgruk replied:
Done!

SuperSandro2000 commented on pkgs/development/python-modules/keyring-24/default.nix:
```diff
@@ -0,0 +1,76 @@
+    license = licenses.mit;
+    mainProgram = "keyring";
+    maintainers = with maintainers; [ jnsgruk ];
+    platforms = platforms.unix;
```
This platforms is already the default

jnsgruk replied:
Removed, thanks.

jnsgruk commented:
Result of `nixpkgs-review pr 325601` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>devpi-client</li>
    <li>devpi-client.dist</li>
  </ul>
</details>
<details>
  <summary>58 packages built:</summary>
  <ul>
    <li>breezy (python312Packages.breezy)</li>
    <li>breezy.dist (python312Packages.breezy.dist)</li>
    <li>cabal2nix</li>
    <li>charmcraft</li>
    <li>charmcraft.dist</li>
    <li>dep2nix</li>
    <li>flatpak-builder</li>
    <li>flatpak-builder.doc</li>
    <li>flatpak-builder.installedTests</li>
    <li>flatpak-builder.man</li>
    <li>hydra_unstable</li>
    <li>luarocks-packages-updater</li>
    <li>nix-prefetch-bzr</li>
    <li>nix-prefetch-scripts</li>
    <li>nix-update-source</li>
    <li>nix-update-source.dist</li>
    <li>python311Packages.breezy</li>
    <li>python311Packages.breezy.dist</li>
    <li>python311Packages.check-manifest</li>
    <li>python311Packages.check-manifest.dist</li>
    <li>python311Packages.craft-application</li>
    <li>python311Packages.craft-application-1</li>
    <li>python311Packages.craft-application-1.dist</li>
    <li>python311Packages.craft-application.dist</li>
    <li>python311Packages.craft-archives</li>
    <li>python311Packages.craft-archives.dist</li>
    <li>python311Packages.craft-providers</li>
    <li>python311Packages.craft-providers.dist</li>
    <li>python311Packages.craft-store</li>
    <li>python311Packages.craft-store.dist</li>
    <li>python311Packages.keyring_24</li>
    <li>python311Packages.keyring_24.dist</li>
    <li>python311Packages.launchpadlib</li>
    <li>python311Packages.launchpadlib.dist</li>
    <li>python312Packages.check-manifest</li>
    <li>python312Packages.check-manifest.dist</li>
    <li>python312Packages.craft-application</li>
    <li>python312Packages.craft-application-1</li>
    <li>python312Packages.craft-application-1.dist</li>
    <li>python312Packages.craft-application.dist</li>
    <li>python312Packages.craft-archives</li>
    <li>python312Packages.craft-archives.dist</li>
    <li>python312Packages.craft-providers</li>
    <li>python312Packages.craft-providers.dist</li>
    <li>python312Packages.craft-store</li>
    <li>python312Packages.craft-store.dist</li>
    <li>python312Packages.keyring_24</li>
    <li>python312Packages.keyring_24.dist</li>
    <li>python312Packages.launchpadlib</li>
    <li>python312Packages.launchpadlib.dist</li>
    <li>rockcraft</li>
    <li>rockcraft.dist</li>
    <li>silver-platter</li>
    <li>silver-platter.dist</li>
    <li>snapcraft</li>
    <li>snapcraft.dist</li>
    <li>vcstool</li>
    <li>vcstool.dist</li>
  </ul>
</details>

natsukium commented on pkgs/development/python-modules/keyring_24/default.nix:
```diff
@@ -0,0 +1,76 @@
+    hash = "sha256-wzJ7b/r8DovvvbWXys20ko/+XBIS92RfGG5tmVeomNs=";
+  };
+
+  nativeBuildInputs = [
+    installShellFiles
+    setuptools-scm
+    shtab
+  ];
```
```suggestion
  nativeBuildInputs = [
    installShellFiles
    shtab
  ];
  
  build-system = [
    setuptools
    setuptools-scm
  ];
```
Could you use `build-system` and `dependencies` instead?
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#buildpythonpackage-parameters-buildpythonpackage-parameters

jnsgruk replied:
Done!

natsukium commented on pkgs/development/python-modules/keyring_24/default.nix:
```diff
@@ -0,0 +1,76 @@
+    shtab
+  ];
+
+  propagatedBuildInputs =
```
```suggestion
  dependencies =
```

jnsgruk replied:
Done!
